### PR TITLE
Raise exception when None value is passed to arrow.get()

### LINKED ISSUE #900: Should arrow.get(None) work?
As per the title should the following work?

```shell
>>> arrow.get(None)
<Arrow [2020-12-20T20:12:12.631217+00:00]>
```

https://github.com/arrow-py/arrow/blob/master/arrow/factory.py#L169

There could be an issue where someone unintentionally passes `None` in but gets no error. Realistically is this used?

### PR DESCRIPTION
**Pull request by Armaxxx**
## Pull Request Checklist

Thank you for taking the time to improve Arrow! Before submitting your pull request, please check all *appropriate* boxes:

- [x] 🧪  Added **tests** for changed code.
- [x] 🛠️  All tests **pass** when run locally (run `tox` or `make test` to find out!).
- [x] 🧹  All linting checks **pass** when run locally (run `tox -e lint` or `make lint` to find out!).
- [ ] 📚  Updated **documentation** for changed code.
- [x] ⏩  Code is **up-to-date** with the `master` branch.

If you have *any* questions about your code changes or any of the points above, please submit your questions along with the pull request and we will try our best to help!

## Description of Changes

Closes: https://github.com/arrow-py/arrow/issues/900

### CODE REVIEW

**Comment by Armaxxx** [comment]
Also I'm not sure how the errors are happening, probably trying to run some None valued get command.

**Comment by Armaxxx** [comment]
I added an assertRaises from unittest

File: arrow/factory.py [hunk 1]
```diff
@@ -167,7 +167,8 @@ def get(self, *args, **kwargs):
 
             # (None) -> now, @ utc.
             if arg is None:
-                return self.type.utcnow()
+                raise NameError
+                # return self.type.utcnow()
```
Comment (jadchaar) [inline arrow/factory.py:171]: A `ValueError` is more appropriate in this situation.

File: tests/test_factory.py [hunk 2]
```diff
@@ -10,7 +11,7 @@
 
 
 @pytest.mark.usefixtures("arrow_factory")
-class TestGet:
+class TestGet(unittest.TestCase):
```
Comment (jadchaar) [inline tests/test_factory.py:14]: No need to use `unittest` here. We are using `pytest`, so all you need to do to test this is:

```python
with pytest.raises(ValueError):
    self.factory.get(None)
```

**Review by jadchaar — CHANGES_REQUESTED** [review]

File: tests/test_factory.py (hunk 2 above)
Comment (Armaxxx) [reply tests/test_factory.py]: How silly of me the example was right below, it has been changed.

File: arrow/factory.py (hunk 1 above)
Comment (Armaxxx) [reply arrow/factory.py]: I had my doubts, but you're completely right, it's been changed.

File: arrow/factory.py [hunk 3]
```diff
@@ -167,7 +167,8 @@ def get(self, *args, **kwargs):
 
             # (None) -> now, @ utc.
             if arg is None:
-                return self.type.utcnow()
+                raise ValueError
```
Comment (krisfremen) [inline arrow/factory.py:170]: I'm OK with ValueError, but wouldn't a TypeError be more indicative?

`TypeError("Cannot parse None.")` perhaps?

**Review by krisfremen — CHANGES_REQUESTED** [review]
Few comments.

File: arrow/factory.py (hunk 3 above)
Comment (jadchaar) [reply arrow/factory.py:170]: After reading [this SO post](https://stackoverflow.com/questions/48343387/valueerror-and-typeerror-in-python), I agree with Kris, this should be a `TypeError`. I think Kris's message is sufficient as well.

File: arrow/factory.py [hunk 4]
```diff
@@ -167,7 +167,8 @@ def get(self, *args, **kwargs):
...
+                # return self.type.utcnow()
```
[the remark this answers is not in the document]
Comment (Armaxxx) [reply arrow/factory.py]: Of course

File: arrow/factory.py [hunk 5]
```diff
@@ -167,7 +167,8 @@ def get(self, *args, **kwargs):
 
             # (None) -> now, @ utc.
```
[the remark this answers is not in the document]
Comment (Armaxxx) [reply arrow/factory.py]: Yeah makes sense, it now says that it raises an exception

File: arrow/factory.py (hunk 3 above)
Comment (Armaxxx) [reply arrow/factory.py]: Great, I used the comment you provided if you don't mind. It's done.

**Comment by Armaxxx** [comment]
Weird, the error is in test-months on humanize class, on the test_arrow.py file, but I didn't fiddle with it.

**Comment by jadchaar** [comment]
@Armaxxx yeah this is due to a recent change to our humanize bounds. @systemcatch ,@krisfremen, and I need to take a closer look (ref https://github.com/arrow-py/arrow/issues/890). Your changes now look good to go though!

**Review by krisfremen — APPROVED** [review]
Looking good.

File: arrow/factory.py [hunk 6]
```diff
@@ -165,9 +165,9 @@ def get(self, *args, **kwargs):
         if arg_count == 1:
             arg = args[0]
 
-            # (None) -> now, @ utc.
+            # (None) -> raises an exception
             if arg is None:
-                return self.type.utcnow()
+                raise TypeError("Cannot parse None.")
```
Comment (systemcatch) [inline arrow/factory.py:170]: Maybe here we should say `"Cannot parse argument of type None"`

**Review by systemcatch — COMMENTED** [review]
Hey @Armaxxx, only one suggestion from me.

**Review by jadchaar — APPROVED** [review]

**Pull request merged**